build.assets/tooling/cmd/changelog/main.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/build.assets/tooling/lib/git"
+)
+
+var (
+	baseBranch = kingpin.Flag(
+		"baseBranch",
+		"The base release branch to generate the changelog for.  It will be of the form branch/v*",
+	).Envar("BASE_BRANCH").String()
+
+	baseTag = kingpin.Flag(
+		"baseTag",
+		"The tag/version to generate the changelog from. It will be of the form vXX.Y.Z, e.g. v15.1.1",
+	).Envar("BASE_TAG").String()
+
+	// eotTimestamp is just a convenience variable to signify no upper bound on searching PR's by time
```

Comment:
What does "eot" stand for? I'm not familiar with the term.

Reply:
Stands for "End Of Time" which I'm now realizing isn't a common term. I must have picked it up from a game or something. I changed it to `timeNow` so it's more obvious.